Code/GraphMol/MolStandardize/Charge.cpp

```diff
@@ -421,10 +415,9 @@ void Uncharger::unchargeInPlace(RWMol &mol) {
     }
 
     // now do the other negative groups if we still have charges left:
-    neg_surplus = a_matched - q_matched;
     if (neg_surplus > 0) {
       boost::dynamic_bitset<> skipChargeSep(mol.getNumAtoms());
-      for (const auto &pair : n_atoms) {
+      for (const auto &pair : a_atoms) {
```

Comment:
This one I think was just a bug, maybe a regression. The test on `skipChargeSep` is meant to apply to `a_atoms`, and `n_atoms` is also not a superset of it.